### PR TITLE
add 2023 Ohio tax rules

### LINKED ISSUE #4194: Add 2023 Ohio Tax Rules
2023 Income Tax Form Instructions: https://dam.assets.ohio.gov/image/upload/tax.ohio.gov/forms/ohio_individual/individual/2023/it1040-sd100-instructionbooklet.pdf

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #4194

### CODE REVIEW

File: policyengine_us/parameters/gov/states/oh/tax/income/agi_threshold.yaml [hunk 1]
```diff
@@ -1,9 +1,15 @@
 description: Ohio requires filers with this adjusted gross income or more to pay individual income tax.
 values:
   2021-01-01: 25_000
+  2022-01-01: 26_050
+  2023-01-01: 26_050
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/agi_threshold.yaml:5]:
```suggestion
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/oh/tax/income/agi_threshold.yaml:5]: dont need to duplicate the value if it says the same - just the reference is enough

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml [hunk 2]
```diff
@@ -1,15 +1,10 @@
 description: Ohio provides a joint filing credit equal to a fraction of tax liability less these non-refundable tax credits.
 values:
   2021-01-01:
-    - oh_cdcc # Line 6
-    - oh_senior_citizen_credit # Line 3
-    - oh_retirement_credit # Line 2
-    - oh_exemption_credit # Line 9
-  2023-01-01:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:8]: no changes? did we miss a credit that was either added in 2023 or revoked?

Not sure why we would do this before

File: policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml [hunk 3]
```diff
@@ -10,21 +10,40 @@ brackets:
...
       2021-01-01: 0.03688
+      2023-01-01: 0.0375
+      2024-01-01: 0.035
   - threshold: 
-      2021-01-01: 110_650
+      values:
+        2021-01-01: 110_650
+        2022-01-01: 115_300
+        2023-01-01: .inf
+        2024-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml:43]:
```suggestion
```

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/lump_sum_distribution/expected_remaining_years.yaml [hunk 4]
```diff
@@ -397,89 +395,132 @@ brackets:
...
+      2022-01-01: 1.4
   - threshold:
       2021-01-01: 112
+      2022-01-01: .inf
     amount:
       2021-01-01: 0.8
+      2022-01-01: 1.4
   - threshold:
       2021-01-01: 113
+      2022-01-01: .inf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/lump_sum_distribution/expected_remaining_years.yaml:511]: could you test this in 2022 / 2023 just to make sure we get expected values for higher thresholds

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml (hunk 2 above)
Comment (hua7450) [reply policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:8]: I don't know how those credits got picked in the list, they are the exact same list comparing 2021 and 2023, so I deleted the 2023 part. We do have a file named "non_refundable_credits.yaml", that file has the adopted credit got revoke in 2023.

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:8]: Makes sense, and in the forms it is all the same?

Comment (hua7450) [reply policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:8]: There are 4 new credits that we have not yet programmed: credit for the beginning farmers financial management program, welcome home Ohio credit, Ohio low-income housing credit, affordable single-family housing credit.  ([line 21,22,32,33](https://dam.assets.ohio.gov/image/upload/tax.ohio.gov/forms/ohio_individual/individual/2023/1040-bundle-original.pdf#page=7))

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:8]: What are the details of:
Ohio low-income housing credit, affordable single-family housing credit?

The other two are very low priority

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml [hunk 5]
```diff
@@ -1,15 +1,10 @@
...
-    - oh_cdcc # Line 6
-    - oh_senior_citizen_credit # Line 3
-    - oh_retirement_credit # Line 2
-    - oh_exemption_credit # Line 9
+    - oh_cdcc
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/joint_filing/other_non_refundable_credits.yaml:4]: could you keep the Line numbers - it is easier to see which credits are meant based on the form

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml [hunk 6]
```diff
@@ -33,5 +51,9 @@ metadata:
   reference:
     - title: Ohio Tax Rate Schedule
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml:52]:
```suggestion
    - title: Section 5747.02 | Tax rates.
```

File: policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml [hunk 7]
```diff
@@ -10,21 +10,39 @@ brackets:
       # is below $25,000.
       2021-01-01: 0.01385
   - threshold: 
-      2021-01-01: 25_000
+      values:
+        2021-01-01: 25_000
+        2022-01-01: 26_050
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml:15]: As the legal code specifies in 2023 before indexing
```suggestion
        2022-01-01: 26_050
        2023-01-01: 26_050
```

File: policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/rates.yaml:1]: add uprating per 5747.02(A)(5)

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**